src/main/java/ui/GUIController.java

```diff
@@ -102,13 +108,38 @@ public void panelFilterExpressionChanged(FilterPanel changedPanel) {
                 .collect(Collectors.toList());
     }
 
+    private void updateRateLimitsEvent(UpdateRateLimitsEvent e) {
+
+        updateAPIBox(e);
+
+        if (e.allowUpdateOfRefreshRate) {
+            updateSyncRefreshRate(e);
+        }
+
+    }
+
+    private void updateSyncRefreshRate(UpdateRateLimitsEvent e) {
```

Comment:
@dariusf feels it would be neater to have a single event handler for this which dispatches into the two methods.

My earlier version is with two different event handler for updateAPIBox and updateSyncRefreshRate.